ludwig/schema/preprocessing.py

```diff
@@ -0,0 +1,751 @@
+        default=True,
+        description="If true, then the number of channels in the dataset is inferred from a sample of the first image "
+        "in the dataset.",
+    )
+
+    infer_image_dimensions: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="If true, then the height and width of images in the dataset will be inferred from a sample of "
+        "the first image in the dataset. Each image that doesn't conform to these dimensions will be "
+        "resized according to resize_method. If set to false, then the height and width of images in the "
+        "dataset will be specified by the user.",
+    )
+
+    infer_image_max_height: Optional[int] = schema_utils.PositiveInteger(
+        default=256,
+        allow_none=False,
+        description="If infer_image_dimensions is set, this is used as the maximum height of the images in "
+        "the dataset.",
+    )
+
+    infer_image_max_width: Optional[int] = schema_utils.PositiveInteger(
+        default=256,
+        allow_none=False,
+        description="If infer_image_dimensions is set, this is used as the maximum width of the images in "
+        "the dataset.",
+    )
+
+    infer_image_sample_size: Optional[int] = schema_utils.PositiveInteger(
+        default=100,
+        allow_none=False,
+        description="The sample size used for inferring dimensions of images in infer_image_dimensions.",
+    )
+
+    scaling: Optional[str] = schema_utils.StringOptions(
+        ["pixel_normalization", "pixel_standardization"],
+        default="pixel_normalization",
+        allow_none=False,
+        description="The scaling strategy for pixel values in the image.",
+    )
+
+    in_memory: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="Defines whether image dataset will reside in memory during the training process or will be "
+        "dynamically fetched from disk (useful for large datasets). In the latter case a training batch "
+        "of input images will be fetched from disk each training iteration.",
+    )
+
+    num_processes: Optional[int] = schema_utils.PositiveInteger(
+        default=1,
+        allow_none=False,
+        description="Specifies the number of processes to run for preprocessing images.",
+    )
+
+
+@dataclass
+@register_preprocessor("audio")
+class AudioPreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+
+    audio_file_length_limit_in_s: Optional[int] = schema_utils.NonNegativeFloat(
+        default=7.5,
+        allow_none=False,
+        description="Float value that defines the maximum limit of the audio file in seconds. All files longer than "
+        "this limit are cut off. All files shorter than this limit are padded with padding_value",
+    )
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        ["fill_with_const", "fill_with_mode", "fill_with_mean", "backfill", "bfill", "pad", "ffill", "drop_row"],
```

Comment:
Since we're including `fill_with_const` in the enum, do we need to include a `fill_value` option as well?